trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/chandler/MemoryHandler.java

```diff
@@ -768,13 +779,26 @@ public ExpressionResult getReadCall(final Expression address, final CType result
 
 	public ExpressionResult getReadCall(final Expression address, final CType resultType, final boolean unchecked) {
 		final ILocation loc = address.getLocation();
+		if (unchecked) {
+			final int byteSize;
+			if (resultType instanceof CPointer) {
+				byteSize = mTypeSizes.getSizeOfPointer();
+			} else if (resultType instanceof final CPrimitive prim) {
+				byteSize = mTypeSizes.getSize(prim.getType());
+			} else {
+				throw new AssertionError("We only support reading primitive or pointer types");
+			}
+			return new ExpressionResult(new RValue(
+					readFromHeap(loc, determineMemoryArrayForType(resultType), address, resultType, byteSize),
+					resultType));
+		}

```

Comment:
Instead of having this parameter, could we not just have a different method?

Reply:
Yes, this should be possible in principle. At some point we probably need a case distinction, but I will check it again.